### PR TITLE
Add scopeflags file

### PR DESCRIPTION
**Pull request by darkHarry**
Thought of adding the scopes and binds into an enum but that would require implementing bitwise or-ing, and-ing, adding traits. So added global constants just like the js file. Let me know if this need to be changed.

### CODE REVIEW

**Comment by ryzokuken** [comment]
https://docs.rs/bitflags/1.2.0/bitflags/

File: Cargo.toml [hunk 1]
```diff
@@ -7,3 +7,4 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+bitflags = "1.2.0"
```
Comment (ryzokuken) [inline Cargo.toml]: newline pls.

File: src/scopeflags.rs [hunk 2]
```diff
@@ -1,22 +1,34 @@
...
-const SCOPE_SUPER: u8 = 64;
-const SCOPE_DIRECT_SUPER: u8 = 128;
+#[macro_use]
+extern crate bitflags;
 
-pub fn function_flags(async: bool, generator: bool) -> u8 {
-    let is_async = if async { SCOPE_ASYNC } else { 0u8 };
-    let is_generator = if generator { SCOPE_GENERATOR } else { 0u8 };
-    SCOPE_FUNCTION | is_async | is_generator
+bitflags! {
```
Comment (ryzokuken) [inline src/scopeflags.rs:4]: 💯

File: src/scopeflags.rs [hunk 3]
```diff
@@ -1,22 +1,34 @@
...
 
-const BIND_NONE: u8 = 0;
-const BIND_VAR: u8 = 1;
-const BIND_LEXICAL: u8 = 2;
-const BIND_FUNCTION: u8 = 3;
-const BIND_SIMPLE_CATCH: u8 = 4;
-const BIND_OUTSIDE: u8 = 5;
\ No newline at end of file
+pub fn function_flags(async: bool, generator: bool) -> Scopes {
+    let is_async = if async { Scopes::SCOPE_ASYNC } else { Scopes::SCOPE_ZERO };
```
Comment (ryzokuken) [inline src/scopeflags.rs]: are you sure this works? `async` is a keyword in both JavaScript and Rust IIRC.

File: src/scopeflags.rs [hunk 4]
```diff
@@ -1,22 +1,34 @@
...
+bitflags! {
+    pub struct Binds: u8 {
+        const BIND_NONE = 0b00000000;
+        const BIND_VAR = 0b00000001;
+        const BIND_LEXICAL = 0b00000010;
+        const BIND_FUNCTION = 0b00000011;
+        const BIND_SIMPLE_CATCH = 0b00000100;
+        const BIND_OUTSIDE = 0b00000101;
+    }
+}
```
Comment (ryzokuken) [inline src/scopeflags.rs]: newline again.

**Review by ryzokuken — CHANGES_REQUESTED** [review]
couple final touches. Thanks!

File: src/scopeflags.rs (hunk 3 above)
Comment (darkHarry) [reply src/scopeflags.rs]: changing it.

**Review by ryzokuken — APPROVED** [review]
Perfect! Thank you so much.

**Pull request merged**